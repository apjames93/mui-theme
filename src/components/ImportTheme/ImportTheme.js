import React, { useState } from 'react';
import PropTypes from 'prop-types';
import base64url from 'base64url';
import { Form, FileInput, MuiSubmit } from 'rff-wrapper';
import {
  Dialog,
  DialogTitle,
  DialogContent,
  DialogContentText,
  DialogActions,
} from '@material-ui/core';
import AttachFileIcon from '@material-ui/icons/AttachFile';
import DialogIconButton from '../sharedComponents/DialogIconButton/DialogIconButton';

export const ImportTheme = ({
  setState,
  theme,
  accessToken,
  pageRedirect,
}) => {
  const [open, setOpen] = useState(false);

  const handleOpen = () => {
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  const onSubmit = async (values) => {
    const { importTheme } = values;
    const base64Split = importTheme.split('data:application/json;base64,');
    const updatedTheme = base64url.decode(base64Split[1]);
    setState({
      accessToken,
      theme: updatedTheme,
      pageRedirect,
    });
    handleClose();
  };

  return (
    <>
      <DialogIconButton title="Import Existing Theme" handleOpen={handleOpen} icon={<AttachFileIcon />} />
      <Dialog open={open} onClose={handleClose}>
        <DialogTitle>Add Storyblok Access Token</DialogTitle>
        <DialogContent>
          <DialogContentText>
            Drag .json file to file input to import theme.
          </DialogContentText>
          <Form
            onSubmit={onSubmit}
            initialValues={{
              theme,
            }}
          >
            <FileInput name="importTheme" acceptFileTypes={['application/json']} />
            <DialogActions>
              <MuiSubmit buttonText="Import Theme" />
            </DialogActions>
          </Form>
        </DialogContent>
      </Dialog>
    </>
  );
};

export default ImportTheme;

ImportTheme.propTypes = {
  accessToken: PropTypes.string.isRequired,
  setState: PropTypes.func.isRequired,
  theme: PropTypes.shape.isRequired,
  pageRedirect: PropTypes.string.isRequired,
};

ImportTheme.defaultProps = {};
