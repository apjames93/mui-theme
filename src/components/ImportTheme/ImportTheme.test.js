import React from 'react';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';
import ImportTheme from './ImportTheme';

function setup() {
  const props = {
    setState: jest.fn(),
    theme: { primary: 'asdf' },
    accessToken: '1234',
    pageRedirect: '/',
  };

  const comp = mount(<ImportTheme {...props} />);
  return { comp, props };
}

describe('<ImportTheme />', () => {
  it('renders ImportTheme', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  it('should handleSubmit and take imported theme value and decode it', () => {
    const { comp } = setup();
    const iconBtn = comp.find('ForwardRef(IconButton)').first();
    console.log(comp.debug());
  });

  it('should handle close and close dialog when submitted or close is clicked', () => {
    const { comp } = setup();
    const dialog = comp.find('WithStyles(ForwardRef(Dialog))').first().props();
    expect(dialog.open).toEqual(false);
    const iconBtn = comp.find('ForwardRef(IconButton)').first();
    iconBtn.simulate('click');
    const dialogOpen = comp.find('WithStyles(ForwardRef(Dialog))').first().props();
    expect(dialogOpen.open).toEqual(true);
    const dialogClose = comp.find('WithStyles(ForwardRef(Dialog))').first().prop('onClose');
    act(() => {
      dialogClose();
    });
    setTimeout(() => {
      const closeDialog = comp.find('WithStyles(ForwardRef(Dialog))').first().props();
      expect(closeDialog.open).toEqual(false);
    }, 800);
  });

  test('snapshot', () => {
    const tree = renderer.create(<ImportTheme />);
    expect(tree).toMatchSnapshot();
  });
});
