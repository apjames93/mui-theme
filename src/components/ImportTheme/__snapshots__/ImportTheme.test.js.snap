// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImportTheme /> snapshot 1`] = `
<button
  aria-describedby={null}
  className="MuiButtonBase-root MuiIconButton-root"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  title="Import Existing Theme"
  type="button"
>
  <span
    className="MuiIconButton-label"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M16.5 6v11.5c0 2.21-1.79 4-4 4s-4-1.79-4-4V5c0-1.38 1.12-2.5 2.5-2.5s2.5 1.12 2.5 2.5v10.5c0 .55-.45 1-1 1s-1-.45-1-1V6H10v9.5c0 1.38 1.12 2.5 2.5 2.5s2.5-1.12 2.5-2.5V5c0-2.21-1.79-4-4-4S7 2.79 7 5v12.5c0 3.04 2.46 5.5 5.5 5.5s5.5-2.46 5.5-5.5V6h-1.5z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;
